src/dataflow/components/nodes/factories/logic-rete-node-factory.tsx

```diff
@@ -0,0 +1,59 @@
+import Rete from "rete";
+import { Node, Socket } from "rete";
+import { NodeData } from "rete/types/core/data";
+import { NumControl } from "../controls/num-control";
+import { DropdownListControl } from "../controls/dropdown-list-control";
+import { NodeOperationTypes } from "../../../utilities/node";
+
+export class LogicReteNodeFactory extends Rete.Component {
+  private numSocket: Socket;
+  constructor(numSocket: Socket) {
+    super("Logic");
+    this.numSocket = numSocket;
```

Comment:
There appears to be some redundancy between the various node factories, e.g. `this.numSocket`. Would it make sense for there to be a `BaseReteNodeFactory` which encapsulated some of the commonality?

Reply:
This could be left for a future enhancement if you don't want to deal with it now.

Reply:
Yeah, I might punt on it for now, but it's not a bad idea since there's overlap.